zstacklib/zstacklib/gpu/vendors/nvidia.py

```
@@ -312,6 +312,113 @@ def _parse_vgpu_output(output):
+            except (ValueError, IndexError):
+                continue
+        return result
+
+    @staticmethod
+    def _get_worker_host_pid(worker):
+        """Get the host-visible PID for a TF worker (handles container mode)."""
+        if worker.container_id:
+            import subprocess
+            proc = None
+            try:
+                proc = subprocess.Popen(
```

Ruff (0.15.11)

[error] `subprocess` call: check for execution of untrusted input

(S603)